### DISCUSSION TITLE
Adonis Sessions

### DISCUSSION THREAD

**Discussion by Hiba-Ed** [adonisjs/core#1577] [Help] [answered]
Hi, I was looking at the [docs](https://adonisjs.com/docs/4.1/sessions) for sessions and I don't understand the code under the view helpers heading. Why are there two separate snippets? Are we saying that the code in both snippets is equivalent or that both of them need to be present to achieve the desired functionality? For instance:

```
session.flashOnly(['username'])
```
```
<input type="text" name="username" value="{{ old('username', '') }}" />
```
Can someone please shed some light on this? TIA.

**Comment by pk4all** [comment, marked as answer]
@Hiba-Ed That both of them need to be present to achieve the desired functionality.
We need where required some information hidden like password or any other info.
